proc-macros/tests/ui/correct/basic.rs

```diff
@@ -0,0 +1,72 @@
+//! Example of using proc macro to generate working client and server.
+
+use jsonrpsee::{
+	proc_macros::rpc,
+	types::async_trait,
+	ws_client::*,
+	ws_server::{SubscriptionSink, WsServerBuilder},
+};
+use std::{net::SocketAddr, sync::mpsc::channel};
+
+#[rpc(client, server, namespace = "foo")]
+pub trait Rpc {
+	#[method(name = "foo")]
+	async fn async_method(&self, param_a: u8, param_b: String) -> u16;
+
+	#[method(name = "bar")]
+	fn sync_method(&self) -> u16;
+
+	#[subscription(name = "sub", unsub = "unsub", item = String)]
```

Comment:
would super useful with a test/example for `subscriptions` with param(s),

I didn't managed to get it working:

```rust
  --> examples/proc_macro.rs:35:1
   |
35 | #[rpc(client, server, namespace = "foo")]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   | |
   | expected enum `jsonrpsee::types::Error`, found enum `CallError`
   | help: try using a variant of the expected enum: `Call(#[rpc(client, server, namespace = "foo")])`
   |
note: return type inferred to be `jsonrpsee::types::Error` here
  --> examples/proc_macro.rs:35:1
   |
35 | #[rpc(client, server, namespace = "foo")]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: this error originates in the attribute macro `rpc` (in Nightly builds, run with -Z macro-backtrace for more info)

```

I suspect that `CallError` is returned but it should be `Error`

Reply:
Sure thing, will add!

Reply:
Must be fixed now.